src/utils.py

```diff
@@ -55,55 +143,59 @@ def get_body(url: str, random_seed: float = None) -> str:
             response_successful = True
         else:
             print('response failure, retrying...')
-            time.sleep(random.random()*3)
+            time.sleep(random.random() * 3)
 
-    #clean up
+    # clean up
     if not response_successful:
-        raise UrlException
+        raise UrlException(" Sorry, we not able to down load this content at the moment.")
+
+    ######################################
+
+    ######################################
+    #           parse response           #
+    ######################################
+
+    site_domain = get_domain(url)
+    if site_domain is None:
+        raise UrlException(" Sorry, something has gone horribly wrong,.")
```

Comment:
Grammar police: unnecessary comma at the end of this line, before the full top.

Reply:
I'll update